src/include/processor/operator/persistent/copy_rel.h

```diff
@@ -46,8 +43,16 @@ class CopyRelSharedState {
     inline void incrementNumRows(common::row_idx_t numRowsToIncrement) {
         numRows.fetch_add(numRowsToIncrement);
     }
+    inline void setNumRows(common::row_idx_t numRows_) { numRows.store(numRows_); }
+    inline common::row_idx_t getNumRows() const { return numRows.load(); }
     inline void updateRelsStatistics() { relsStatistics->setNumTuplesForTable(tableID, numRows); }
-
+    inline common::offset_t getNextRelOffset(transaction::Transaction* transaction) const {
+        return relsStatistics->getRelStatistics(tableID, transaction)->getNextRelOffset();
+    }
+    inline std::vector<std::unique_ptr<common::LogicalType>>& getColumnTypesRef() {
```

Comment:
Based on our previous discussion, should we avoid use unique_ptr on LogicalType?

Reply:
Yep. I will remove it from storage in another PR.